src/cpp/r/session/RStdCallbacks.cpp

```diff
@@ -445,7 +445,13 @@ void RBusy(int which)
 {
    try
    {
-      s_callbacks.busy(which == 1) ;
+      // synchronize locale whenever R busy state changes
+      // (done relatively eagerly to ensure synchronization
+      // happens on each REPL iteration after user code is run)
+      r::util::synchronizeLocale();
```

Comment:
This requires the `base` package to be attached -- can R ever be busy before then?

It feels like this should be hooked up to OnDetectChanges with `ChangeSourceREPL` instead. Does that seem right?

Reply:
Okay, fair enough!